demo/main.js

```diff
@@ -1482,6 +1492,18 @@ shakaDemo.Main = class {
     }
     params.push('uilang=' + this.getUILocale());
 
+    const preferredVideoCodecs = /** @type {!Array.<string>} */(
```

Comment:
I would say that you should instead change `runThroughHashParams` to properly handle arrays config values, but that would probably overcomplicate this PR.
You could at least do:

```js
for (const key of ['preferredVideoCodecs', 'preferredAudioCodecs']) {
  const array = /** @type {!Array.<string>} */(this.getCurrentConfigValue(key));
  if (array.length) {
    params.push(key + '=' + array.join(','));
  }
}
```

Reply:
Done!